jcl/src/java.base/share/classes/java/lang/Class.java

```diff
@@ -4707,11 +4711,22 @@ public boolean isRecord() {
 	 * Returns an array of RecordComponent objects for a record class.
 	 * 
 	 * @return array of RecordComponent objects, one for each component in the record.
-	 * For a class that is not a record an empty array is returned.
+	 * For a class that is not a record, null is returned.
 	 * For a record with no components an empty array is returned.
+	 * 
+	 * @throws SecurityException 
 	 */
+	@CallerSensitive
 	public RecordComponent[] getRecordComponents() {
-		throw new InternalError("Compile stub invoked! For JEP 359 support see https://github.com/eclipse/openj9/pull/7946"); //$NON-NLS-1$
+		SecurityManager security = System.getSecurityManager();
+		if (security != null) {
+			ClassLoader callerClassLoader = ClassLoader.getStackClassLoader(1);
+			checkMemberAccess(security, callerClassLoader, Member.DECLARED);
+		}
+
+		return getRecordComponentsImpl();
```

Comment:
Yes that case is handled in `getRecordComponentsImpl`. My thought was since `isRecord()` is also a native it would make more sense to have only one native call here instead of two.

Reply:
It's, after jitting, a FastJNI that will run quickly.  Much cheaper than the `getRecordComponentsImpl` JNI call